### PR TITLE
feat(breadcrumb): revamping the breadcrumb and navigation

### PR DESCRIPTION
**Pull request by bdebon**
CleanShot 2022-11-02 at 16 54 16@2x

This PR finally does a little bit more that what it was supposed to:
- DarkModeEnabler so we can use tailwind `dark:` utils
- Breadcrumb darkmode theme
- Breadcrumb new version
- Fix a bug about filters on environment variable when you create a new env variable (it was keeping the filter on the header when the table was not filtered anymore)
- Tests
- New contrast for the whole interface

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(breadcrumb): revamping the breandcrump and navigation - 2022-11-02T15:54:49Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/74f60df4-bd36-4b4f-bf02-6816341a453d/applications)

 Another comment will be posted when deployments are terminated

**Comment by evoxmusic** [comment]
Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://z74f60df4-z86925b05-gtw.za8ad0659.bool.sh)
:point_right: [storybook](https://z74f60df4-z31f89a57-gtw.za8ad0659.bool.sh)

File: libs/pages/environments/src/lib/ui/container/container.tsx [hunk 1]
```diff
@@ -24,9 +28,10 @@ export interface ContainerProps {
 
 export function Container(props: ContainerProps) {
   const { children } = props
-  const { organizationId, projectId } = useParams()
+  const { organizationId, projectId = '' } = useParams()
```
Comment (RemiBonnet) [inline libs/pages/environments/src/lib/ui/container/container.tsx:31]: You don't need `organizationId = ''` ?

File: libs/shared/ui/src/lib/components/layouts/breadcrumb/breadcrumb.tsx [hunk 2]
```diff
@@ -217,44 +224,45 @@ export function BreadcrumbMemo(props: BreadcrumbProps) {
...
+                  menuItems={environmentMenu}
+                  paramId={environmentId}
+                  link={SERVICES_URL(organizationId, projectId, environmentId) + SERVICES_GENERAL_URL}
+                />
+                {(applicationId || databaseId) && (
+                  <>
+                    <div className="w-4 h-auto text-element-light-lighter-600 text-center mx-3">/</div>
+                    <div className="flex items-center">
+                      <BreadcrumbItem
+                        isLast={true}
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/layouts/breadcrumb/breadcrumb.tsx:246]: You can add directly `isLast`

**Review by RemiBonnet — APPROVED** [review]
Well done @bdebon I added small comments, happy to see a clean dark mode, you can merge !! :)

**Pull request merged**